### PR TITLE
Add documentation for resource.rb fields

### PR DESCRIPTION
**Pull request by rileykarson**
/cc @rambleraptor @slevenick in case I misrepresented any of the fields Terraform doesn't use.

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: api/resource.rb [hunk 1]
```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
     module Properties
       include Api::Object::Named::Properties
 
+      # [Required] A description of the resource that's surfaced in provider
```
Comment (rambleraptor) [inline api/resource.rb]: The validate methods show the required/optional. Do we still think we need them in descriptions?

File: api/resource.rb [hunk 2]
```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
...
+      # [Optional] The HTTP verb used during update. Defaults to :PUT.
+      attr_reader :update_verb
+      # [Optional] The HTTP verb used during delete. Defaults to :DELETE.
+      attr_reader :delete_verb
+
+      # ====================
+      # Collection / Identity URL Configuration
+      # ====================
+      #
+      # [Optional] This is the type of response from the collection URL. It
```
Comment (rambleraptor) [inline api/resource.rb]:
```suggestion
      # [Optional] (Api::Resource::ResponseList) This is the type of response from the collection URL. It
```
What do you think about this? The type information would be easy to find + in a standardized place.

File: api/resource.rb (hunk 1 above)
Comment (rileykarson) [reply api/resource.rb]: Yes- the moment you start reading the validate methods you see hard to understand blocks like this which distract from that kind of information.

```ruby
      check :nested_query, type: Api::Resource::NestedQuery
      if @nested_query&.is_list_of_ids && @identity&.length != 1
        raise ':is_list_of_ids = true implies resource`\
              `has exactly one :identity property"'
      end
```

File: api/resource.rb (hunk 2 above)
Comment (rileykarson) [reply api/resource.rb]: Done

File: api/resource.rb [hunk 3]
```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
...
+      attr_reader :update_mask
+      # [Optional] If set to true, the object has a `self_link` field. This is
+      # typical of older GCP APIs.
+      attr_reader :has_self_link
+
+      # ====================
+      # URL / HTTP Configuration
+      # ====================
+      #
+      # [Optional] The "identity" URL of the resource.. Defaults to:
```
Comment (danawillow) [inline api/resource.rb]: nit:
```suggestion
      # [Optional] The "identity" URL of the resource. Defaults to:
```

File: api/resource.rb [hunk 4]
```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
...
+      # * collection url when the create_verb is :POST
```
Comment (danawillow) [inline api/resource.rb]: This is a bit confusing since collection url hasn't been defined yet- I think base url accomplishes the same thing

**Pull request merged**